Formspec: Add support for vertical scrollbar in textarea

adelcoding1 opened the pull request:
- textarea Formspec uses now a `GUIEditBoxWithScrollBar` instead of the default `irr::gui::IGUIEditBox`
- `GUIEditBoxWithScrollBar` allows to use a vertical scrollbar when it's needed
- `GUIEditBoxWithScrollBar` implements `irr::gui::IGUIEditBox`

jnumm committed:
Update .appdata and .desktop files

Fixer-007 committed:
README.txt: Clarify loading of minetest.conf

sofar committed:
Font: attempt fallback font, abort if no fonts found.

If you happen to have a font_path setting that is incorrect,
minetest will just attempt to start the gui without a valid
font which leads to a segfault later on.

We can attempt to load the fallback font path fairly easy,
but if that fails we should give up with a proper error message
and not a weird segfault later. This forces an abort() if
the fallback fails as well, and prints a useful error
message to the console.

paramat committed:
Dungeongen: Add and improve parameters

Add:
Bool for 'only_in_ground'.
Min and max corridor length.
Min and max room size with X, Y, Z components.
Min and max large room size with X, Y, Z components.

'only_in_ground = false' allows core mapgens to create structures
in air and water using dungeongen.
Corridor length parameters replace a fixed random range.
Room size parameters replace the former system where one parameter
'roomsize' was added to fixed random ranges.

All parameters are set for no change to current dungeon behaviour.

Remove some now-redundant and long-unused code.

adelcoding1 committed:
Formspec: Add support for vertical scrollbar in textarea

adelcoding1 committed:
Formspec: Add support for vertical scrollbar in textarea

nerzhul commented on build/android/jni/Android.mk:
```diff
@@ -149,6 +149,7 @@ LOCAL_SRC_FILES := \
 		jni/src/genericobject.cpp                 \
 		jni/src/gettext.cpp                       \
 		jni/src/guiChatConsole.cpp                \
+		jni/src/guieditboxwithscrollbar.cpp       \
```
please use same filename as other gui*.cpp files

nerzhul commented on src/guieditboxwithscrollbar.h:
```diff
@@ -0,0 +1,180 @@
+	wchar_t m_passwordchar;
+	EGUI_ALIGNMENT m_halign, m_valign;
+
+	std::vector< core::stringw > m_broken_text;
```
no space in template types

nerzhul commented on src/guieditboxwithscrollbar.cpp:
```diff
@@ -0,0 +1,1484 @@
+		{
+		case EET_GUI_EVENT:
+			if (event.GUIEvent.EventType == EGET_ELEMENT_FOCUS_LOST)
+			{
```
please fix the code style and place your braces on upper lines for conditions, switch & for

nerzhul commented on src/guieditboxwithscrollbar.cpp:
```diff
@@ -0,0 +1,1484 @@
+
+	// control shortcut handling
+
+	if (event.KeyInput.Control)	{
```
missing space

SmallJoker commented on src/guiFormSpecMenu.cpp:
```diff
@@ -1153,8 +1154,8 @@ void GUIFormSpecMenu::parseTextArea(parserData* data,
-#endif
-			e = Environment->addEditBox(spec.fdefault.c_str(), rect, true, this, spec.fid);
+#endif			
+			e = new GUIEditBoxWithScrollBar(spec.fdefault.c_str(), true, Environment,this, spec.fid, rect, 20);
```
Tailing indents after `#endif`
Missing space after `true, Environment`

Ekdohibs commented:
Since this is a modification of irrlicht code, maybe it would be better to put it in the corresponding folder.

red-001 committed:
Fix crash that can be caused by the shutdown command. (#5292)

Bond-009 committed:
Fixed spelling mistakes (#5312)

paramat committed:
Dungeons: Use 'block' instead of 'brick' for nodebox stairs

Affects only sandstone dungeons. Nodebox stairs made from
'sandstone_block' look better because every step is undivided.

adelcoding1 committed:
FormSpec: Add position and anchor elements (#5284)

zaoqi committed:
Add minetest.spawn_falling_node(pos) (#5339)

* Add minetest.spawn_falling_node(pos)

* lua_api.txt: Add minetest.spawn_falling_node(pos)

* Update minetest.spawn_falling_node(pos)

nerzhul committed:
CI: Add memleak checking using valgrind (#5350)

Add a new step to check memleaks on our current unit tests suite

adelcoding1 committed:
Merge branch 'textarea_with_scrollbar' of https://github.com/adelcoding1/minetest into textarea_with_scrollbar

adelcoding1 committed:
Add an auto/optional vertical scrollbar in textarea

nerzhul commented:
Seems your merged instead of rebase, please do git rebase master

kilbith commented:
New PR here : https://github.com/minetest/minetest/pull/5355
